web_external/js/models/MongoSourceModel.js

```diff
@@ -0,0 +1,23 @@
+minerva.models.MongoSourceModel = minerva.models.SourceModel.extend({
+
+    initialize: function () {
+    },
+
+    createSource: function (params) {
+        girder.restRequest({
+            path: '/minerva_source_mongo',
+            type: 'POST',
+            data: params,
+            error: null // don't do default error behavior (validation may fail)
+        }).done(_.bind(function (resp) {
+            this.set(resp);
+            this.trigger('m:sourceReceived');
+        }, this)).error(_.bind(function (err) {
+            this.trigger('m:error', err);
+            // If we couldn't successfully start the s3 import destroy the item.
+            this.destroy();
```

Comment:
You don't need to do this explicit destroy call.  

The S3 Source created an additional folder to store the contents of the files imported from S3, and the destroy call of the Girder Item needed to be overridden so that the additional folder was deleted if there was an error creating the S3 source.